cf-serverd/server_tls.c

```diff
@@ -429,7 +429,7 @@ int ServerTLSSessionEstablish(ServerConnectionState *conn)
         int ret = SSL_accept(ssl);
         if (ret <= 0)
         {
-            TLSLogError(ssl, LOG_LEVEL_ERR,
+            (void)TLSLogError(ssl, LOG_LEVEL_ERR,
```

Comment:
So the reason is for documentation towards human programmers who read the code; Why not add a comment?
Answer: Because if we start commenting on each and every function it will become a mess. Same if we start typecasting each and every call that we don't check its value. Too many functions like that, ugly mess.

`TLSLogError()` might not be necessary to check, so skip all casts - and I say /might/ because I haven't validated this refactoring yet. However `strlcpy()` should **almost always** be checked. If not checked, then a `(void)` cast is **not enough** documentation; I want to read a comment saying **why** truncation is not checked. So I don't see a point in `(void)` casting at all.

C is an ugly language, but very powerful. Type casts are one of the most powerful and most dangerous features of C: they nullify the type system, they shut the compiler up from any warning. Lets use them only where we *must*. From HACKING.md:

> Avoid using type casts, unless absolutely necessary. Usually a compiler warning is better satisfied with correct code rather than using a type cast.